assets/search/components/WatchListEditor.jsx

```diff
@@ -0,0 +1,259 @@
+
+        this.props.saveWatchList(this.state.watchList);
+    }
+
+    render() {
+        const {item, isAdmin, user} = this.props;
+        const watchList = get(this.state, 'watchList') || {};
+        const propsToFields = {
+            'onChange': this.onChange,
+            'readOnly': !isAdmin,
+        };
+        const subscribed = (get(this.state, 'watchList.users') || []).includes(user);
+
+        return (
+            <div className='list-item__preview'>
+                <div className="list-item__preview-header">
+                    <h3>{get(item, 'name')}</h3>
+                    <button
+                        id="hide-sidebar"
+                        type="button"
+                        className="icon-button"
+                        onClick={this.props.closeEditor}
+                        disabled={this.state.saving}
+                    >
+                        <i className="icon--close-thin icon--gray" />
+                    </button>
+                </div>
+                <ul className='nav nav-tabs'>
+                    {this.tabs.filter((tab, index) => index === 0 || this.props.item._id).map((tab) => (
+                        <li key={tab.name} className='nav-item'>
+                            <a
+                                name={tab.name}
+                                className={`nav-link ${this.state.activeTab === tab.name && 'active'}`}
+                                href='#'
+                                onClick={this.handleTabClick}>{tab.label}
+                            </a>
+                        </li>
+                    ))}
+                </ul>
+                <div className="list-item__preview-form">
+                    <div className='tab-content'>
+                        {this.state.activeTab === 'watch-list' &&
+                        <div className='tab-pane active' id='watch-list'>
+                            <form>
+                                <div className="list-item__preview-form">
+                                    {!isAdmin && (
+                                        <div className='form-group'>
+                                            <input
+                                                name='notifications'
+                                                type="button"
+                                                className="btn btn-outline-primary"
+                                                value={subscribed ? gettext('Unsubscribe') : gettext('Subscribe')}
+                                                onClick={this.onChange}
+                                                disabled={this.state.saving} />
+                                        </div>
+                                    )}
+                                    <TextInput
+                                        name='subject'
+                                        label={gettext('Subject line')}
+                                        value={watchList.subject}
```

Comment:
If `watchList` is null then we should not be rendering the input elements until it has a value. It is causing the following warning in React:

`Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components`